pedsnetdcc/prepdb.py

```diff
@@ -0,0 +1,244 @@
+    :type: str
+    :param db_type: 'prod' or 'internal'
+    :type: str
+    :return: SQL statements
+    :rtype: list(str)
+    :raises: ValueError
+    """
+    if db_type == 'internal':
+        tmpl = _sql_site_internal_template
+    elif db_type == 'prod':
+        tmpl = _sql_site_prod_template
+    else:
+        raise ValueError('Invalid db_type: {}'.format(db_type))
+
+    sql = tmpl.replace('{{.Site}}', site)
+    return [x for x in sql.split("\n") if x]
+
+
+_sql_vocab_internal = """
+grant all                  on schema vocabulary to loading_user;
+grant usage                on schema vocabulary to achilles_user, dqa_user, pcor_et_user, harvest_user, peds_staff;
+grant select on all tables in schema vocabulary to achilles_user, dqa_user, pcor_et_user, harvest_user, peds_staff;
+alter default privileges for role loading_user in schema vocabulary grant select on tables to achilles_user, dqa_user, pcor_et_user, harvest_user, peds_staff;
+"""
+
+_sql_vocab_prod = """
+grant usage                on schema vocabulary to harvest_user, peds_staff;
+grant select on all tables in schema vocabulary to harvest_user, peds_staff;
+alter default privileges for role dcc_owner in schema vocabulary grant select on tables to harvest_user, peds_staff;
+"""
+
+
+def _vocabulary_sql(db_type):
+    """Return a list of statements to create the vocabulary schema.
+
+    :param db_type: 'prod' or 'internal'
+    :type: str
+    :rtype: list(str)
+    :raises: ValueError
+    """
+    sql = "create schema if not exists vocabulary authorization dcc_owner;\n"
+
+    if db_type == 'internal':
+        sql += _sql_vocab_internal
+    elif db_type == 'prod':
+        sql += _sql_vocab_prod
+    else:
+        raise ValueError('Invalid db_type: {}'.format(db_type))
+    return [x for x in sql.split("\n") if x]
+
+
+def prepare_database(model_version, conn_str, db_type, update=False,
+                     dcc_only=False):
+    """Create a new database containing vocabulary and site schemas.
+
+    The initial conn_str is used for issuing a CREATE DATABASE statement.
+    statement.
+
+    A subsequent connection to the newly created database is established
+    in order to create schemas.
+
+    :param model_version: PEDSnet model version, e.g. X.Y.Z
+    :type: str
+    :param conn_str: libpq connection string
+    :type: str
+    :param db_type: 'internal' or 'prod'
+    :type: str
+    :param update: assume the database is already created
+    :type: bool
+    :param dcc_only: only create schemas for `dcc` (no sites)
+    :return: None, or (if `psql` is True) a SQL string
+    :rtype: str
+    """
+
+    stmts = []
+
+    database_name = _make_database_name(model_version)
+
+    if not update:
+        stmts.extend(_create_database_sql(database_name))
+
+    stmts.extend(_database_privileges_sql(database_name))
+
+    logger.info({'msg': 'Starting database preparation.'})
+
+    with psycopg2.connect(conn_str) as conn:
+        conn.set_isolation_level(0)
```

Comment:
If there were an `isolation_level` argument to the `db.py` functions, this function could use those and not need to worry about error handling or messaging properly, right?